api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/CounterHandler.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.api.jaxrs;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.hawkular.metrics.core.api.MetricsService.DEFAULT_TENANT_ID;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Response;
+
+import org.hawkular.metrics.api.jaxrs.callback.NoDataCallback;
+import org.hawkular.metrics.api.jaxrs.callback.SimpleDataCallback;
+import org.hawkular.metrics.core.api.Counter;
+import org.hawkular.metrics.core.api.MetricsService;
+
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+
+/**
+ * @author Stefan Negrea
+ *
+ */
+@Path("/")
+@Consumes(APPLICATION_JSON)
+@Produces(APPLICATION_JSON)
+@Api(value = "", description = "Availability metrics interface")
+public class CounterHandler {
```

Comment:
Return codes are not mentioned for documentation purposes. Also, are these counters still hidden=true (they were hidden in the older versions because they weren't ready for use) ?